03_letters/nicholascross/main.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, strings.Join(sortLetters(letters("aba")), "\n"))
+}
+
+func sortLetters(m map[rune]int) []string {
+	keys := make([]string, 0, len(m))
+	for k, v := range m {
+		keys = append(keys, fmt.Sprintf("%c:%d", k, v))
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+func letters(s string) map[rune]int {
+	count := map[rune]int{}
+
```

Comment:
Little nit, I would allocate the space here and remove the additional line.

```go
count := make(map[rune]int)
for _, char := range s {
  count[char]++
}
return count
```

Reply:
@rokane I am confused, what is the difference between these statements?

```
map[rune]int{}
```

```
make(map[rune]int)
```

I thought they both initialise an empty map?

Reply:
Do we prefer one style of initialisation over the other, is that it?

Reply:
I've just been reading up on this, the two result in the same. I am happy for you to leave it either way, I usually always declare it with the `make` keyword, but it seems totally fine to do use the `{}` as well. Alternatively you could declare the maximum size of the map upfront which might give a slight performance bump, but in this context not required.
```go
make(map[rune]int, len(s))
```
[What is the difference between make and {} to initialise a map](https://groups.google.com/forum/#!topic/golang-nuts/YnqSi2yURSY)